src/items/functions.md

```diff
@@ -203,12 +203,33 @@ extern "C" fn new_i32() -> i32 { 0 }
 let fptr: extern "C" fn() -> i32 = new_i32;
 ```
 
-Functions with an ABI that differs from `"Rust"` do not support unwinding in the
-exact same way that Rust does. Therefore, unwinding past the end of functions
-with such ABIs causes the process to abort.
+### Unwinding
+
+Most ABI strings come in two variants, one with an `-unwind` suffix and one without.
+The `Rust` ABI always permits unwinding, so there is no `Rust-unwind` ABI.
+
+The choice of ABI, together with the [panic mode][panic-modes], determines the
+behavior when unwinding out of a function.
+
+In the table below, "Foreign unwind (unforced)" refers to something like a C++
```

Comment:
It feels a little strange to call out "Foreign unwind (unforced)" but make no mention of presumably *forced* unwinding as a distinct section. Is that UB? How does one determine whether unwinding is forced or unforced?

Reply:
Forced unwinding happens on for example pthread_cancel. It is like unforced unwinding except that you can't catch it and depending in the platform cleanup may or may not happen. If I recall correctly a forced unwind passing through a rust stack frame which has locals to drop on unwinding is UB, but passing through a rust stack frame without any locals to drop hasn't been decided as UB or not yet.

Reply:
Ah, yeah, this is a bit of an oddity; the phrasing & column header are from RFC 2945, but since we're not actually specifying behavior for forced unwind yet, I forgot to include an explanation.

Forced unwinding is `pthread_cancel` in glibc and `longjmp` in Windows. I'm not aware of any other uses. So for the purpose of defining language behavior, we effectively want to say "here are the circumstances in which `phtread_cancel` and `longjmp` are safe _on any platform_", and that will imply the behavior of forced unwind on platforms where it's used.

I'm not sure whether it's preferable to explain here what forced unwinding is or just remove "unforced" from the text for now. I think the latter is probably okay, since the explanation of "something like a C++ exception" already implies that this isn't talking about forced unwinding.

Reply:
I think we should specify as precisely as possible the exceptions we are including (and aren't including) - "like C++" doesn't feel like the right level of detail here.

Reply:
@Mark-Simulacrum @bjorn3 @Amanieu @nbdd0121 

Do you have any specific suggestions for how to define "native unforced unwinding" here, in an appropriate level of detail? I believe the salient points to capture are:

* It's determined by the platform (compiler backend + OS + libc implementation if applicable + architecture + ...?)
* It specifically doesn't include "forced unwinding", terminology that we didn't invent but which...seems kind of hard to find a good reference for? Everything I know about forced unwinding was told to me as part of this project, I think mostly by one of you.

Reply:
I think it's fine to just say that this is defined per target and is usually whatever C++ uses for unwinding on that target. And we can just omit any mention of forced unwinding.

Reply:
@Mark-Simulacrum Are you okay with Amanieu's suggestion?

Reply:
Defined per target is fine, but I feel like we would still *want* at least some targets to have actual definitions. That definition can be "we match C++ behavior".

Reply:
@Mark-Simulacrum @Amanieu How about:

> Native unwinding is defined per-target. On targets that support throwing and catching C++ exceptions, it refers to the mechanism used to implement this feature.

Reply:
Is it acceptable to link to [the RFC](https://rust-lang.github.io/rfcs/2945-c-unwind-abi.html#forced-unwinding)? I don't see any other links to that site in the reference.